staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go

```diff
@@ -1631,7 +1631,9 @@ func RunTestListContinuation(ctx context.Context, t *testing.T, store storage.In
 		}
 	}
 	options := storage.ListOptions{
-		ResourceVersion: "0",
+		// Limit is ignored when ResourceVersion is set to 0.
+		// Set it to consistent read.
+		ResourceVersion: "",
```

Comment:
The failures from before this PR could be easily triggered by adding `time.Sleep(time.Second)` in appropriate tests (cacher_test.go) after creating watchcache but before starting the test.

Reply:
I'm sure we had one but I can't find any now. We should add that as a follow-up (though I wouldn't block this PR on it as it's not regressing our coverage).

Reply:
okay, I can add the new test if you don't have time. Just let me know.

Reply:
So I found what I was looking for:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go#L862-L866
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go#L881-L885

We are now ignoring it, it might be better to slightly update them to check that we return everything in those cases.
If you have time to take it, it would be great.

Reply:
sure, i will have a look.